pkg/operator/configobservation/node/observe_latency_profile.go

```diff
@@ -0,0 +1,127 @@
+	}
+
+	// read the observed value
+	var observedDefaultNotReadyTolerationSeconds string
+	switch {
+	case node.Spec.WorkerLatencyProfile == configV1.DefaultUpdateDefaultReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.DefaultNotReadyTolerationSeconds)
+	case node.Spec.WorkerLatencyProfile == configV1.MediumUpdateAverageReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.MediumNotReadyTolerationSeconds)
+	case node.Spec.WorkerLatencyProfile == configV1.LowUpdateSlowReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.LowNotReadyTolerationSeconds)
+	// in case of empty worker latency profile, do not update config
+	case node.Spec.WorkerLatencyProfile == "":
+		return existingConfig, errs
+	}
+
+	// read the current value
+	var currentDefaultNotReadyTolerationSeconds string
+	currentDefaultNotReadyTolerationSecondsSlice, _, err := unstructured.NestedStringSlice(
+		existingConfig, notReadyTolerationSecondsPath...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to extract not ready toleration seconds from the existing config: %v", err))
+		// keep going, we are only interested in the observed value which will overwrite the current configuration anyway
+	}
+	if len(currentDefaultNotReadyTolerationSecondsSlice) > 0 {
+		currentDefaultNotReadyTolerationSeconds = currentDefaultNotReadyTolerationSecondsSlice[0]
+	}
+
+	// see if the current and the observed value differ
+	observedConfig := map[string]interface{}{}
+	if currentDefaultNotReadyTolerationSeconds != observedDefaultNotReadyTolerationSeconds {
+		if err = unstructured.SetNestedStringSlice(observedConfig,
+			[]string{observedDefaultNotReadyTolerationSeconds},
+			notReadyTolerationSecondsPath...); err != nil {
+			return existingConfig, append(errs, err)
+		}
+		return observedConfig, errs
+	}
+	// nothing has changed return the original configuration
+	return existingConfig, errs
+}
+
+// ObserveUnreachableTolerationSeconds observes the value that should be set for default-unreachable-toleration-seconds
+// api server argument on the basis of provided worker latency profile from config node object.
+func ObserveUnreachableTolerationSeconds(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains default-not-ready-toleration-seconds field.
+		ret = configobserver.Pruned(ret, unreachableTolerationSecondsPath)
+	}()
+
+	nodeLister := genericListers.(configobservation.Listers).NodeLister
+	node, err := nodeLister.Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// we got an error so without the node object we are not able to determine worker latency profile
+		return existingConfig, append(errs, err)
+	}
+
+	// read the observed value
+	var observedDefaultUnreachableTolerationSeconds string
+	switch {
```

Comment:
consider handling unsupported/future values

Reply:
Added a default case for now. In the future if more profile(s) have to be added their respective arguments have to be included in this observer and controller anyway.